pbsmetrics/metrics.go

```diff
@@ -168,6 +169,7 @@ func RequestStatuses() []RequestStatus {
 		RequestStatusErr,
 		RequestStatusNetworkErr,
 		RequestStatusBlacklisted,
+		RequestStatusQueueTimeout,
 	}
```

Comment:
I think we should also add a metric to track the queue duration to give us better visibility into the performance. That'll give us an idea for how long wait times are looking in the queue and could power decisions for scaling or adjusting the queue window. Thoughts?

Reply:
Agree! Please see the above comment.

Reply:
Will move to separate metrics struct

Reply:
To add a new metric to track the queue time (and possible outcome, such as timeouts) you'll need to add a method to the `MetricsEngine` interface and fill in the functionality for the Influx and Prometheus real engines, the "fake" ones for testing (MetricsEngineMock, DummyMetricsEngine), and the MultiMetricsEngine multiplexor.

Reply:
Yes, and fix unit tests.